packages/langium/src/workspace/workspace-lock.ts

```diff
@@ -28,8 +28,13 @@ export interface WorkspaceLock {
      * If a new write action is queued up while a read action is waiting, the write action will receive priority and will be handled before the read action.
      *
      * Note that read actions are not allowed to modify anything in the workspace. Please use {@link write} instead.
+     *
+     * @param action the action to perform.
+     * @param priority
+     * If set to true, the action will receive immediate attention and will be instantly executed. All other queued read/write actions will be postponed.
+     * This should only be used for actions that require a specific document state and have already awaited the necessary write actions.
      */
-    read<T>(action: () => MaybePromise<T>): Promise<T>;
+    read<T>(action: () => MaybePromise<T>, priority?: boolean): Promise<T>;
```

Comment:
I am associating `priority` with an ordering/number. I understand the usage here as `hasPriority`. Maybe there is a better name, like: `isUrgent`, `important`, `handleFirst`...